includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -845,27 +845,14 @@ public function prepare_source( $user_id, $force_save_source = false, $existing_
 			$customer->maybe_create_customer();
 
 			// Check if the customer opted to save the payment method to file.
-			$maybe_saved_card    = isset( $_POST[ 'wc-' . $payment_method . '-new-payment-method' ] ) && ! empty( $_POST[ 'wc-' . $payment_method . '-new-payment-method' ] );
-			$is_sepa_source      = isset( $source_object->type ) && 'sepa_debit' === $source_object->type;
-			$save_payment_method = $force_save_source || ( $user_id && $this->saved_cards && $maybe_saved_card );
-
-			/**
-			 * Save the SEPA source to the customer if force save source is true or the user is logged in and the user has opted to save the payment method.
-			 *
-			 * We only need to save SEPA sources here. Saving card payment methods are handled by setup intents later in the flow. @see WC_Gateway_Stripe::process_payment().
-			 */
-			if ( $is_sepa_source && $save_payment_method ) {
-				$response = $customer->attach_source( $source_object->id );
+			$maybe_saved_card = isset( $_POST[ 'wc-' . $payment_method . '-new-payment-method' ] ) && ! empty( $_POST[ 'wc-' . $payment_method . '-new-payment-method' ] );
 
-				if ( ! empty( $response->error ) ) {
-					throw new WC_Stripe_Exception( print_r( $response, true ), $this->get_localized_error_message_from_response( $response ) );
-				}
-				if ( is_wp_error( $response ) ) {
-					throw new WC_Stripe_Exception( $response->get_error_message(), $response->get_error_message() );
+			if ( $force_save_source || ( $user_id && $this->saved_cards && $maybe_saved_card ) ) {
+				$response = $this->maybe_attach_source_to_customer( $source_object, $customer );
+				if ( $response ) {
```

Comment:
If `$response` is used here just to check if the process was successful, I would replace the return with a boolean.

Reply:
👍 changed in [1615f92](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3246/commits/1615f927047bf6b48728d2ae60c7b8384a0916cc)